File: 30 Days of Code/Day 17 More Exceptions/Solution.cs
using System;

class Calculator {
    int power( int n, int p ) {
        if( n < 0 || p < 0 ) {
            throw new Exception( "n and p should be non-negative" );
        }

        return ( int )Math.Pow( n, p );
    }
}

public class Solution {
    public static void Main( String[ ] args ) {
        int T = Convert.ToInt32( Console.ReadLine( ) );
        Calculator calculator = new Calculator( );

        for( int idx = 0; idx < T; idx++ ) {
            string[ ] num = Console.ReadLine( ).Split( );
            int n = int.Parse( num[ 0 ] );
            int p = int.Parse( num[ 1 ] );
            try {
                int result = calculator.power( n, p );
                Console.WriteLine( result );
            } catch( Exception ex ) {
                Console.WriteLine( ex.Message );
            }
        }

    }
}

File: 30 Days of Code/Day 17 More Exceptions/Solution.py
import sys

class Calculator:
    def power( self, n, p ):
        if n < 0 or p < 0:
            raise ValueError( "n and p should be non-negative" )
        return n**p

T = int( input( ).strip( ) )
calculator = Calculator( )
for _ in range( T ):
    n,p = map( int, input( ).split( ) )
    try:
        result = calculator.power( n, p )
        print( result )
    except Exception as e:
        print( e )